staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -260,7 +296,21 @@ func (w *watchCache) processEvent(event watch.Event, resourceVersion uint64, upd
 	// This is safe as long as there is at most one call to processEvent in flight
 	// at any point in time.
 	if w.eventHandler != nil {
-		w.eventHandler(watchCacheEvent)
+		// We copy the event to avoid storing serializations in-memory for a longer
+		// time period. If we could tollerate higher memory usage, we can get rid of it.
```

Comment:
a comment that explains this is setting up caching only for this dispatch of this event would be helpful.

calling out the memory cost of adding caching to the retained list of watch events is good, but it would be helpful to explain what the benefit of that would be (cached encodings for watches started from old versions) and why we don't think that is worth the memory cost

Reply:
done